plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java

```diff
@@ -308,4 +315,22 @@ public static void wrapListenerToHandleSearchIndexNotFound(Exception e, ActionLi
         }
     }
 
+    public static boolean IsIndependentNode(ClusterService clusterService, Settings settings) {
+        isIndependentNode = ML_COMMONS_INDEPENDENT_NODE.get(settings);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(ML_COMMONS_INDEPENDENT_NODE, it -> isIndependentNode = it);
+        return isIndependentNode;
+    }
+
+    public static String getTenantID(ClusterService clusterService, Settings settings, RestRequest restRequest) {
+        if (IsIndependentNode(clusterService, settings)) {
+            String tenantId = restRequest.getHeaders().get(Constants.TENANT_ID).get(0);
+            if (tenantId == null) {
+                throw new OpenSearchStatusException("Tenant ID can't be null", RestStatus.INTERNAL_SERVER_ERROR);
+            }
+            return tenantId;
+        } else {
+            return null;
```

Comment:
What should be the flow when it is not an independent node and null is returned here?

Reply:
for single tenant case, tenant id will be null and that will be stored in the index as null.